x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1830,7 +1852,10 @@ private static Expression typeSpecificConvert(ConvertFunction convert, Source so
                 originalFieldAttr.id(),
                 true
             );
-            Expression e = ((Expression) convert).replaceChildren(Collections.singletonList(resolvedAttr));
+            Expression fn = (Expression) convert;
+            List<Expression> children = new ArrayList<>(fn.children());
+            children.set(0, resolvedAttr);
+            Expression e = ((Expression) convert).replaceChildren(children);
```

Comment:
Why did we make this change? AbstractConvertFunction contains only a single child, the field. Or do we have something that does not extend that class?

Reply:
OK. I see FromAggregateMetricDouble does not extend AbstractConvertFunction.